### PR TITLE
feat(experiments): holdout groups UI

### PR DESCRIPTION
**Pull request by jurajmajerik**
## Changes

- Add UI for managing holdout groups
- Add the holdout to test variants in the results calculation

### 1. List holdouts
image

### 2. Create/edit holdout
image

### 3. Edit holdout during experiment creation
image

### 4. Edit holdout in the Draft mode
image

### 5. Show holdout in the Launched mode (no longer editable)
image

### 6. Show results for the holdout
image

## How did you test this code?
👀

### CODE REVIEW

File: frontend/src/scenes/experiments/Holdouts.tsx [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+                            />
+                            <LemonInput
+                                data-attr="rollout-percentage"
+                                type="number"
+                                className="ml-2 mr-1.5 max-w-30"
+                                value={holdout.filters?.[0]?.rollout_percentage || 100}
+                                onChange={(rollout_percentage) =>
+                                    setHoldout({
+                                        filters: {
+                                            groups: [{ properties: [], rollout_percentage }],
```
Comment (neilkakkar) [inline frontend/src/scenes/experiments/Holdouts.tsx:192]: groups seem extra

**Review by neilkakkar — COMMENTED** [review]
haven't tried running locally yet, not sure if you've incorporated changes with my latest `holdout_id` pr.

File: frontend/src/scenes/experiments/ExperimentView/components.tsx [hunk 2]
```diff
@@ -52,8 +52,23 @@ export function VariantTag({
 }): JSX.Element {
     const { experimentResults, getIndexForVariant } = useValues(experimentLogic({ experimentId }))
 
+    if (variantKey.startsWith('holdout-')) {
```
[the remark this answers is not in the document]
Comment (jurajmajerik) [reply frontend/src/scenes/experiments/ExperimentView/components.tsx]: I think this check should be good enough for now: https://github.com/PostHog/posthog/pull/25772/files#diff-ac20870bd5fba7ae95bc9fde57e9108900e26f210382221cc1466059172748f3R55

Can add validation later if we run into issues

**Comment by jurajmajerik** [comment]
Feedback addressed :)

**Review by neilkakkar — APPROVED** [review]

**Pull request merged**